src/segments/repeat.c

```diff
@@ -201,14 +231,16 @@ int repeat_segment_set(uint32_t field, void *value, struct mixed_segment *segmen
     break;
   case MIXED_REPEAT_MODE:
     if(*(enum mixed_repeat_mode *)value != MIXED_RECORD &&
-       *(enum mixed_repeat_mode *)value != MIXED_PLAY){
+       *(enum mixed_repeat_mode *)value != MIXED_PLAY &&
+       *(enum mixed_repeat_mode *)value != MIXED_RECORD_ONCE){
       mixed_err(MIXED_INVALID_VALUE);
       return 0;
     }
     data->mode = *(enum mixed_repeat_mode *)value;
     switch(data->mode){
     case MIXED_RECORD: segment->mix = repeat_segment_mix_record; break;
     case MIXED_PLAY: segment->mix = repeat_segment_mix_play; break;
+    case MIXED_RECORD_ONCE: segment->mix = repeat_segment_mix_record_once; break;
```

Comment:
You should reset the buffer_index here, as otherwise the recording duration for record_once will be unpredictable.

Reply:
I didn't reset it on purpose - mode can be changed during mixing and it doesn't necessarily makes sense to reset buffer position.

Probably functions like repeat_segment_reset and repeat_segment_frame_position should be introduced instead?

Reply:
There's functionally only two ways to use this segment:

1. Keep recording until at some point you decide to play back the last N seconds. In this case record_once is useless and it doesn't ever matter what the internal frame position is.
2. Record a buffer now and then play it back. In this case record_once is the only useful option, and only if it resets the internal state once you switch to record_once. Otherwise the length of time until playback is unpredictable and essentially random, defeating the point.

The additional complexity from exposing the  internal state isn't useful in either case.

One thing we *should* do though is this:

When switching away from record or record_once, apply a low pass filter at the current sample_index to ensure the looping point does not cause crackling.